api/api_tests/internal/primitives/nim/model_interface/test_yolox_interface_base.py

```diff
@@ -75,22 +75,14 @@ def setUp(self):
         self.postprocess_results_patcher = patch(f"{MODULE_UNDER_TEST}.postprocess_results")
         self.mock_postprocess_results = self.postprocess_results_patcher.start()
 
-        self.image_patcher = patch(f"{MODULE_UNDER_TEST}.Image")
-        self.mock_image = self.image_patcher.start()
-        self.mock_pil_image = MagicMock()
-        self.mock_image.fromarray.return_value = self.mock_pil_image
-        self.mock_pil_image.size = (200, 100)
-
-        # Mock BytesIO and base64
-        self.bytesio_patcher = patch(f"{MODULE_UNDER_TEST}.io.BytesIO")
-        self.mock_bytesio = self.bytesio_patcher.start()
-        self.mock_buffer = MagicMock()
-        self.mock_bytesio.return_value = self.mock_buffer
-        self.mock_buffer.getvalue.return_value = b"test_image_data"
-
-        self.base64_patcher = patch(f"{MODULE_UNDER_TEST}.base64.b64encode")
-        self.mock_base64 = self.base64_patcher.start()
-        self.mock_base64.return_value = b"base64_encoded_data"
+        # Mock numpy_to_base64 and scale_image_to_encoding_size functions
+        self.numpy_to_base64_patcher = patch(f"{MODULE_UNDER_TEST}.numpy_to_base64")
+        self.mock_numpy_to_base64 = self.numpy_to_base64_patcher.start()
+        self.mock_numpy_to_base64.return_value = "base64_encoded_data"
+
+        self.scale_image_patcher = patch(f"{MODULE_UNDER_TEST}.scale_image_to_encoding_size")
+        self.mock_scale_image = self.scale_image_patcher.start()
+        self.mock_scale_image.return_value = ("scaled_base64_data", (200, 100))
```

Comment:
We no longer use io.BytesIO and base64.b64encode directly but use them via the numpy_to_base64 instead to ensure the code for image transform is centralized.